diddiparser2/parser.py

```diff
@@ -178,33 +177,63 @@ def parse_string_indexing(self, text):
             final_line = f"{final_line}{true_value}{index[1]}"
         return final_line
 
+    def resolve_value_or_variable(self, arg):
+        """
+        This resolves between direct values or variables
+        using the DSGP 2 specification.
+        """
+        if arg in EXECUTION_VARIABLES.keys():
+            return EXECUTION_VARIABLES[arg]
+        return self.identify_value(arg)
+
     def execute_func(self, line):
         "Run a call(argument) function."
         parsed_line = line.replace(");", "")
         call = parsed_line.split("(")[0]
         pos = len(f"{call}(")  # use this to avoid conflicts
-        arg = parsed_line[pos:]
-        del pos  # delete the aux
-        if arg.startswith("'") or arg.startswith('"'):
-            arg = arg[1:]
-        if arg.endswith("'") or arg.endswith('"'):
-            arg = arg[:-1]
-        arg = self.parse_string_indexing(arg)
+        args_aux = parsed_line[pos:]

```

Comment:
The `DiddiParser.execute_def` method resulted to be more complex after this PR... :/ Maybe in the future we can break this method into a bunch of methods.